smoke-test/tests/cypress/cypress/e2e/domains/nested_domains.js

```diff
@@ -1,53 +1,93 @@
+    }
 
-    it("create a domain, move under parent, remove domain", () => {
-        // Create a new domain without a parent
-        cy.loginWithCredentials();
-        cy.goToDomainList();
-        cy.clickOptionWithTestId("domains-new-domain-button");
-        cy.get('[data-testid="create-domain-name"]').click().type(domainName);
-        cy.get('[data-testid="create-domain-description"]').click().type(domainDescription);
-        cy.clickOptionWithTestId("create-domain-button");
-        cy.waitTextVisible(domainName);
+const moveDomaintoRootLevel = () => {
+    cy.clickOptionWithText(domainName);
+    cy.openThreeDotDropdown();
+    cy.clickOptionWithTestId("entity-menu-move-button");
+    cy.get('[data-testid="move-domain-modal"]').contains("Marketing").click({force: true});
+    cy.waitTextVisible('Marketing')
+    cy.clickOptionWithTestId("move-domain-modal-move-button")
+ }
 
-        // Ensure the new domain has no parent in the navigation sidebar
-        cy.waitTextVisible(domainDescription);
+const moveDomaintoParent = () => {
+    cy.get('[data-testid="domain-list-item"]').contains("Marketing").prev().click();
+    cy.clickOptionWithText(domainName);
+    cy.openThreeDotDropdown();
+    cy.clickOptionWithTestId("entity-menu-move-button");
+    cy.clickOptionWithTestId("move-domain-modal-move-button")
+        }
 
-        // Move a domain from the root level to be under a parent domain
-        cy.clickOptionWithText(domainName);
-        cy.openThreeDotDropdown();
-        cy.clickOptionWithTestId("entity-menu-move-button");
-        cy.get('[data-testid="move-domain-modal"]').contains("Marketing").click({force: true});
-        cy.get('[data-testid="move-domain-modal"]').contains("Marketing").should("be.visible");
-        cy.clickOptionWithTestId("move-domain-modal-move-button").wait(5000);
+ const deleteFromDomainDropdown = () => {
+    cy.clickOptionWithText('Filters')
+    cy.openThreeDotDropdown();
+    cy.clickOptionWithTestId("entity-menu-delete-button");
+    cy.waitTextVisible("Are you sure you want to remove this Domain?");
+    cy.clickOptionWithText("Yes");
+ }       
 
-        // Wnsure domain is no longer on the sidebar navigator at the top level but shows up under the parent
+const deleteDomain = () => {
+    cy.clickOptionWithText(domainName).waitTextVisible('Domains');
+    deleteFromDomainDropdown()
+  }
+
+//Delete Unecessary Existing Domains
+ const deleteExisitingDomain = () => {
+    cy.get('a[href*="urn:li"] span[class^="ant-typography"]')
+          .should('be.visible')
+          .its('length')
+          .then((length) => {
+        for (let i = 0; i < length - 1; i++) {
+            cy.get('a[href*="urn:li"] span[class^="ant-typography"]')
+                .should('be.visible')
+                .first()
+                .click({ force: true });
+            deleteFromDomainDropdown();
+        }
+    });   
+    cy.waitTextVisible('My custom domain'); 
```

Comment:
Why is this supposed to be here at the end of deleting all domains?

Reply:
It's a validation that waits for the text to become visible. 
The text pertains to the Marketing domain, 
which is a single domain already present in domain list
and remains unchanged during the test execution.